lorawan-device/src/region/dynamic_channel_plans/mod.rs

```diff
@@ -144,42 +144,63 @@ impl<R: DynamicChannelRegion> RegionHandler for DynamicChannelPlan<R> {
         }
     }
 
-    fn handle_link_adr_channel_mask(
-        &mut self,
+    fn channel_mask_get(&self) -> ChannelMask<9> {
+        self.channel_mask.clone()
+    }
+
+    fn channel_mask_set(&mut self, channel_mask: ChannelMask<9>) {
+        self.channel_mask = channel_mask;
+    }
+
+    fn channel_mask_update(
```

Comment:
I don't understand what this fn does. what is mask, what is channel_mask. what is self in this context

Reply:
Previously we had wrong behavior which updated existing channel plan (`self.channel_mask`) but there was no way to roll back changes in case the channel plan we ended up was erroneous.

Now, before this function gets called, we create a copy of existing channel plan and pass it into this function as `&mut mask`. Two other arguments come from LinkADRReq. Ideally we could rename the arguments in this way:
*  `mask (`&mut ChannelMask<9>` -> `channel_mask`
* `channel_mask_control` (`u8`) -> `control`
* `channel_mask` (`ChannelMask<2>` -> `mask`
And `&self` could be dropped as well.

Reply:
Ok, updated the signature of this method. Although I'm not 100% satisfied with it yet...